src/profile-logic/perf-script-profile-format.js

```diff
@@ -0,0 +1,225 @@
+    // eg, "java 12688 [002] 6544038.708352: cpu-clock:"
+    // eg, "V8 WorkerThread 25607 4794564.109216: cycles:"
+    // eg, "java 24636/25607 [000] 4794564.109216: cycles:"
+    // eg, "java 12688/12764 6544038.708352: cpu-clock:"
+    // eg, "V8 WorkerThread 24636/25607 [000] 94564.109216: cycles:"
+    // other combinations possible
+    // pattern: thread-name-with-optional-spaces tid time: NNN cycles:XXXX:
+    // alternate pattern: thread-name-with-optional-spaces pid/tid time: NNN cycles:XXXX:
+    // eg, "java 25607/25608 4794564.109216: 33 cycles:uppp"
+    //   (generate with "perf script -F +pid")
+    //
+    // Regexes are evil, and need descriptions (and if really in a string, would need lots of extra escapes).
+    // Match a threadname, with optional spaces, by requiring it to be followed by either a numeric tid or tid/pid
+    // '/^' + '(S.*?)(?=(?:s+(?:(?:d+)/*(?:d+)*s)))s+' +
+    // Match a numeric tid, or a numeric tid/pid
+    // '(d+)/*(d+)*s+([d.]+)' + '/';
+    const sampleStartMatch = /^(\S.*?)(?=(?:\s+(?:(?:\d+)\/*(?:\d+)*\s)))\s+(\d+)\/*(\d+)*\s+([\d.]+)/.exec(
+      sampleStartLine
+    );
+    if (sampleStartMatch) {
+      const threadName = sampleStartMatch[1];
+      const tid = sampleStartMatch[3] || sampleStartMatch[2];
+      const pid = sampleStartMatch[3] ? Number(sampleStartMatch[2]) : 0;
+      const timeStamp = parseFloat(sampleStartMatch[4]) * 1000;
+      // Assume start time is the time of the first sample
+      if (startTime === 0) {
+        startTime = timeStamp;
+      }
+      // Parse the stack frames of the current sample in a nested loop.
+      const stack = [];
+      while (lineIndex < lines.length) {
+        const stackFrameLine = lines[lineIndex++];
+        if (stackFrameLine.trim() === '') {
+          // Sample ends.
+          break;
+        }
+
+        // 	         23fe921 _ZN7mozilla3ipc14MessageChannel11MessageTask3RunEv (/home/mstange/Desktop/firefox/libxul.so)
+        const stackFrameMatch = /^\s*(\w+)\s*(.+) \((\S*)\)/.exec(
+          stackFrameLine
+        );
+        if (stackFrameMatch) {
+          // const pc = stackFrameMatch[1];
+          let rawFunc = stackFrameMatch[2];
+          // const mod = stackFrameMatch[3];
+
+          // Linux 4.8 included symbol offsets in perf script output by default, eg:
+          // 7fffb84c9afc cpu_startup_entry+0x800047c022ec ([kernel.kallsyms])
+          // strip these off:
+          rawFunc = rawFunc.replace(/\+0x[\da-f]+$/, '');
+
+          if (rawFunc.startsWith('(')) {
+            continue; // skip process names
+          }
+
+          stack.push(rawFunc);
+        }
+      }
+
+      if (stack.length !== 0) {
+        stack.reverse();
+        _addThreadSample(pid, tid, threadName, timeStamp, stack);
+      }
+    }
+  }
+
+  const threadArray = Array.from(threadMap.values()).map(thread =>
+    thread.finish()
+  );
+
+  return {
+    meta: {
+      abi: 'x86_64-gcc3',
```

Comment:
Sorry if I'm re-reviewing existing code. It seems like we shouldn't hard-code platform specific values here, because then if we expose this information in the UI, it will be lying to us.

Reply:
I think this one should still be fixed before merging, as it breaks the rule of "DevTools telling lies". I'd be fine if we passed just blank strings for things we don't know.